solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -973,29 +969,31 @@ public void load() {
           solrCores.markCoreAsLoading(cd);
         }
         if (cd.isLoadOnStartup()) {
-          futures.add(
-              coreLoadExecutor.submit(
-                  () -> {
-                    SolrCore core;
-                    try {
-                      if (zkSys.getZkController() != null) {
-                        zkSys.getZkController().throwErrorIfReplicaReplaced(cd);
-                      }
-                      solrCores.waitAddPendingCoreOps(cd.getName());
-                      core = createFromDescriptor(cd, false, false);
-                    } finally {
-                      solrCores.removeFromPendingOps(cd.getName());
-                      if (asyncSolrCoreLoad) {
-                        solrCores.markCoreAsNotLoading(cd);
-                      }
-                    }
-                    try {
-                      zkSys.registerInZk(core, true, false);
-                    } catch (RuntimeException e) {
-                      SolrException.log(log, "Error registering SolrCore", e);
-                    }
-                    return core;
-                  }));
+          coreLoadExecutor.submit(
```

Comment:
Do we only need to do this in an executor if `asyncSolrCoreLoad`? Or if this is async should it use `coreContainerWorkExecutor`?

I'm a little surprised by looking at this diff there are two executors - `coreLoadExecutor` and `coreContainerWorkExecutor`?